src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -547,6 +552,11 @@ def path_to_dataset(self, path):
 
         return ds_name
 
+    def child_dataset_names(self, path):
+        # return child datasets given a dataset `path`.
+        with libzfs.ZFS() as zfs:
```

Comment:
We should catch `ZFSException` and raise it appropriately as a `CallError` like we do for other usages in zfs plugin so that if it needs to be handled for any reason in a middleware plugin, it can be done in a normalized manner ensuring isolated usages of `py-libzfs`.